### ISSUE TITLE
Platform should reject attempts to configure conflicting streams

### ISSUE THREAD

**Issue by edgao** [airbytehq/airbyte#21343]
## Tell us about the problem you're trying to solve
It's currently possible to have a connection where multiple source streams write into the same destination table. For example:
1. In Postgres, create two schemas (`schema1` and `schema2)`
2. In both schemas, create a table `foo_table`
3. Create a connection from Postgres into any DB destination, using the `Destination default` namespace
   ![image](https://user-images.githubusercontent.com/5741425/212152010-9fd2c887-9424-427c-9f37-1c4f7d7e7109.png)
4. Now both source streams will attempt to write into `<destination default>.foo_table`, which causes an error in the destination connector.

This problem also happens if you use a custom namespace format that doesn't include `${SOURCE_NAMESPACE}`.

## Describe the solution you’d like
Fail as early as possible, i.e. display an error when the user tries to configure their connection rather than when they run a sync.

## Describe the alternative you’ve considered or used
Trigger a config error at sync time (https://github.com/airbytehq/airbyte/pull/21342) - this is better and more informative, but still less than ideal.

It could also be useful to enable the `${SOURCE_NAMESPACE}` variable in the stream prefix field, though we'd still need to detect the error first.

## Additional context
seen in https://github.com/airbytehq/oncall/issues/1306

**Comment by davinchia** [comment]
@edgao should this be closed?

**Comment by edgao** [comment]
no, it's under active discussion in https://airbytehq-team.slack.com/archives/C02U6CXJQFN/p1683919222272929

**Comment by nataliekwong** [comment]
Sharing here the related issue on allowing the stream prefix to allow variable values, similarly to how we allow this for the namespace: https://github.com/airbytehq/airbyte/issues/5999
